tests/test_transform_inline.py

```diff
@@ -607,3 +609,145 @@ def test_inline_member_routines_with_associate(frontend):
 
     assocs = FindNodes(Associate).visit(routine.body)
     assert len(assocs) == 2
+
+
+@pytest.mark.parametrize('frontend', available_frontends(xfail=[(OMNI, 'No header information in test')]))
+def test_inline_marked_subroutines(frontend):
+    """ Test subroutine inlining via marker pragmas. """
+
+    fcode_driver = """
+subroutine test_pragma_inline(a, b)
+  use util_mod, only: add_one, add_a_to_b
+  implicit none
+
+  real(kind=8), intent(inout) :: a(3), b(3)
+  integer, parameter :: n = 3
+  integer :: i
+
+  do i=1, n
+    !$loki inline
+    call add_one(a(i))
+  end do
+
+  !$loki inline
+  call add_a_to_b(a(:), b(:), 3)
+
+  do i=1, n
+    call add_one(b(i))
+  end do
+
+end subroutine test_pragma_inline
+    """
+
+    fcode_module = """
+module util_mod
+
+contains
+  subroutine add_one(a)
+    real(kind=8), intent(inout) :: a
+    a = a + 1
+  end subroutine add_one
+
+  subroutine add_a_to_b(a, b, n)
+    real(kind=8), intent(inout) :: a(:), b(:)
+    integer, intent(in) :: n
+
```

Comment:
missing `integer :: i`, which changes the inlined code as well 

```fortran
  DO add_a_to_b_i=1,add_a_to_b_n
    a(add_a_to_b_i) = a(add_a_to_b_i) + b(add_a_to_b_i)
  END DO
```

Thus still works and is valid, just the question whether this should be fixed for possible future changes?

Reply:
Yes, original Fortran should be valid (technically it is, because there's also no `implicit none`)

Reply:
Ahh, this is what the `allowed_aliases` is for; I've amended the test to use this and retrofitted the `implicit none`.